st2common/st2common/models/db/__init__.py

```diff
@@ -123,14 +83,24 @@ def get_all(self, *args, **kwargs):
     def count(self, *args, **kwargs):
         return self.model.objects(**kwargs).count()
 
-    @process_null_filter
-    @process_datetime_ranges
-    def query(self, *args, **kwargs):
-        offset = int(kwargs.pop('offset', 0))
-        limit = kwargs.pop('limit', None)
+    def query(self, offset=0, limit=None, order_by=None, exclude_fields=None,
```

Comment:
This kwargs abuse and decorator magic was really awful. It's really hard to debug and follow what's going on. We should avoid kwargs abuse and such magic - remember, we are writing code for people and not for computers.